### PR TITLE
Fixed the way backslashes are handled in "id" and "class" CSS selectors

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Branch? | 2.3+ |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18247 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/CssSelector/Node/ClassNode.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ class ClassNode extends AbstractNode
     public function __construct(NodeInterface $selector, $name)
     {
         $this->selector = $selector;
-        $this->name = $name;
+        $this->name = str_replace('\\', '', $name);
```
Comment (fabpot) [inline src/Symfony/Component/CssSelector/Node/ClassNode.php]: What `:` as mentioned in the ticket. Can we find the related part of the spec talking about that just to be sure that we are covering all possibilities here?

**Comment by javiereguiluz** [comment]
This is what [the official CSS spec](https://www.w3.org/TR/CSS21/syndata.html#value-def-identifier) says:

---

In CSS, identifiers (including element names, classes, and IDs in selectors) can contain only the characters [a-zA-Z0-9] and ISO 10646 characters U+00A0 and higher, plus the hyphen (-) and the underscore (_); they cannot start with a digit, two hyphens, or a hyphen followed by a digit. Identifiers can also contain escaped characters and any ISO 10646 character as a numeric code. For instance, the identifier "B&W?" may be written as "B\&W\?" or "B\26 W\3F".

---

I'm not sure how to properly fix this issue. Can anyone help me with this? Thanks!

**Comment by xabbuh** [comment]
Looks like we would have to decode numeric character codes too, right?

**Comment by javiereguiluz** [comment]
I'm closing this because I'm lost and I prefer to let someone else fix this. Thanks!

**Pull request closed without merging**